File: skypydb/database/operations/vector/collections/delete_collection.py
"""Collection deletion mixin for vector storage."""

from skypydb.security.validation import InputValidator


class DeleteCollectionMixin:
    """Provide collection table deletion behavior."""

    def delete_collection(self, name: str) -> None:
        """Drop a collection table and remove its metadata entry."""

        name = InputValidator.validate_table_name(name)
        if not self.collection_exists(name):
            raise ValueError(f"Collection '{name}' not found")

        cursor = self.conn.cursor()
        cursor.execute(f"DROP TABLE [vec_{name}]")
        cursor.execute("DELETE FROM _vector_collections WHERE name = ?", (name,))
        self.conn.commit()

File: skypydb-js/src/database/operations/vector/collections/delete_collection.ts
import { CollectionNotFoundError } from "../../../../errors";
import { InputValidator } from "../../../../security/validation";
import { CountItemsMixin } from "./count_items";

export abstract class DeleteCollectionMixin extends CountItemsMixin {
  delete_collection(name: string): void {
    const validated = InputValidator.validate_table_name(name);
    if (!this.collection_exists(validated)) {
      throw new CollectionNotFoundError(`Collection '${validated}' not found`);
    }

    this.conn.prepare(`DROP TABLE [vec_${validated}]`).run();
    this.conn
      .prepare("DELETE FROM _vector_collections WHERE name = ?")
      .run(validated);
  }
}
